File: ga/4omini_result_2024_7/result/1968C_cpp_mini/34_1968C.cpp
use std::io;

const N: usize = 500;
const A: i32 = 1_000_000_000;

fn proc() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: usize = input.trim().parse().expect("Failed to parse");

    let mut x = vec![0; N];
    let mut a = vec![0; N];

    input.clear();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let numbers: Vec<i32> = input.trim().split_whitespace()
        .map(|s| s.parse().expect("Failed to parse number"))
        .collect();

    // Fill the x array with the input numbers
    for i in 1..n {
        x[i] = numbers[i - 1]; // Adjusted index since we start from x[1]
    }

    a[0] = x[1] + 1;
    
    for i in 1..n {
        for j in 0..=A {
            a[i] = x[i] + j * a[i - 1];

            if (if i + 1 < n { x[i + 1] } else { 1 }) < a[i] && a[i] != a[i - 1] {
                break;
            }
        }
    }

    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let t: usize = input.trim().parse().expect("Failed to parse number");

    for _ in 0..t {
        proc();
    }
}

File: ga/gpt3_5_result_2024_6/result/1968C_cpp_gpt/34_1968C.rs
use std::io::{self, BufRead};

const N: usize = 500;
const X: usize = 500;
const A: usize = 1000000000;

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines().map(|line| line.unwrap());

    let t: usize = lines.next().unwrap().parse().unwrap();
    
    for _ in 0..t {
        let n: usize = lines.next().unwrap().parse().unwrap();
        let mut x: Vec<usize> = vec![0; n];
        let mut a: Vec<usize> = vec![0; n];

        let input: Vec<usize> = lines.next().unwrap().split_whitespace().map(|x| x.parse().unwrap()).collect();
        for i in 1..n {
            x[i] = input[i - 1];
        }

        a[0] = x[1] + 1;
        for i in 1..n {
            let mut j = 0;
            while j <= A {
                a[i] = x[i] + j * a[i - 1];
                if (i + 1 < n && x[i + 1] < a[i]) || (a[i] != a[i - 1]) {
                    break;
                }
                j += 1;
            }
        }

        for i in 0..n {
            print!("{} ", a[i]);
        }
        println!();
    }
}